src/components/voting/VoterButtons.tsx

```diff
@@ -97,54 +121,25 @@ const VoterButton = ({
 }
 
 type VoterButtonsProps = VoterProps & {
-  only?: 'Upvote' | 'Downvote',
+  only?: ReactionEnum,
 }
 
 export const VoterButtons = (props: VoterButtonsProps) => {
-  const { post: initialPost, only, ...voterProps } = props
-  const myAddress = useMyAddress()
+  const { post, only, ...voterProps } = props
   const reloadPost = useGetReloadPost()
-  const [ reactionState, setReactionState ] = useState<Reaction>()
-  const [ reloadTrigger, setReloadTrigger ] = useState(true)
-
-  // TODO use redux + subscribe
-  const post = useSubscribedPost(initialPost)
+  const args = { id: post.id }
+  const reaction = useAppSelector(state => selectPostReactionByPostId(state, post.id))
 
-  useSubsocialEffect(({ substrate }) => {
-    let isMounted = true
-
-    async function reloadReaction () {
-      if (!myAddress) return
-
-      // TODO use redux
-      const reactionId = await substrate.getPostReactionIdByAccount(myAddress, idToPostId(post.id))
-      if (!isMounted) return
-
-      const reaction = await substrate.findReaction(reactionId)
-      isMounted && setReactionState(reaction)
-    }
-
-    reloadReaction().catch(err => log.error(
-      'Failed to load a reaction on post with id',
-      post.id, 'by account', myAddress, err
-    ))
-
-    return () => { isMounted = false }
-  }, [ reloadTrigger, myAddress, post.id ])
-
-  const renderVoterButton = (reactionType: ReactionType) => <VoterButton
-    reaction={reactionState}
-    reactionType={reactionType}
-    onSuccess={() => {
-      setReloadTrigger(!reloadTrigger)
-      reloadPost({ id: post.id })
-    }}
+  const renderVoterButton = useCallback((reactionType: ReactionEnum) => <VoterButton
+    reaction={reaction}
+    reactionEnum={reactionType}
+    onSuccess={() => reloadPost(args)}
     post={post}
     {...voterProps}
-  />
+  />, [ post.id, reaction?.kind || 'None' ])
```

Comment:
Why it does not depend on `reaction id` as well?
Or depend on stringified reaction.